File: Getting Started - WCF RIA Services/[C#]-Getting Started - WCF RIA Services/C#/HRApp/Helpers/ResourceWrapper.cs
namespace HRApp.Resources
{
    using System;
    using System.ComponentModel;
    using System.Globalization;
    using System.Windows.Browser;

    /// <summary>
    ///     Wraps access to the strongly typed resource classes so that you can bind
    ///     control properties to resource strings in XAML
    /// </summary>
    public sealed class ResourceWrapper
    {
        private static ApplicationStrings applicationStrings = new ApplicationStrings();
        private static SecurityQuestions securityQuestions = new SecurityQuestions();

        public ApplicationStrings ApplicationStrings
        {
            get { return applicationStrings; }
        }

        public SecurityQuestions SecurityQuestions
        {
            get { return securityQuestions; }
        }
    }
}

File: Getting Started - WCF RIA Services/[VB]-Getting Started - WCF RIA Services/VB/HRApp/Helpers/ResourceWrapper.vb
Imports System
Imports System.ComponentModel
Imports System.Globalization
Imports System.Windows.Browser


''' <summary>
''' Wraps accessing string resources in a way that Bindings can be set through 
''' XAML (which is not supported when binding to a resource class directly 
''' since they have no public constructors)
''' </summary>
Public NotInheritable Class ResourceWrapper
    Private Shared _applicationStrings As New ApplicationStrings()
    Private Shared _securityQuestions As New SecurityQuestions()

    Public ReadOnly Property ApplicationStrings() As ApplicationStrings
        Get
            Return _applicationStrings
        End Get
    End Property

    Public ReadOnly Property SecurityQuestions() As SecurityQuestions
        Get
            Return _securityQuestions
        End Get
    End Property
End Class
